packages/dotcom-build-base/src/index.ts

```diff
@@ -0,0 +1,54 @@
+import type webpack from 'webpack'
+import path from 'path'
+import { CleanWebpackPlugin } from 'clean-webpack-plugin'
+import CompressionPlugin from 'compression-webpack-plugin'
+import ManifestPlugin from 'webpack-assets-manifest'
+
+export class PageKitBasePlugin {
+  apply(compiler: webpack.Compiler) {
```

Comment:
I don't really understand the `apply(compiler)` bit but I'm guessing it's a standard Webpack thing. Do you think you could you explain what it's doing for my benefit?

Reply:
it's a standard Webpack thing. when Webpack loads, it goes through the `plugins` array and calls their `apply` method with the instance of the compiler (the compiler is what you get if you call `webpack()` programatically instead of using the CLI). 

- https://webpack.js.org/contribute/writing-a-plugin/#basic-plugin-architecture
- https://github.com/webpack/webpack/blob/webpack-4/lib/webpack.js#L46-L54

Reply:
so, this isn't "doing" anything, it's defining what Webpack should call when loading